deepedit/ignite/train.py

```diff
@@ -0,0 +1,460 @@
+        )
+    else:
+        network = DynUNet(
+            spatial_dims=3,
+            in_channels=len(labels) + 1,
+            out_channels=len(labels),
+            kernel_size=[[3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3]],
+            strides=[[1, 1, 1], [2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 1]],
+            upsample_kernel_size=[[2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 1]],
+            norm_name="instance",
+            deep_supervision=False,
+            res_block=True,
+        )
+    return network
+
+
+def get_pre_transforms(labels, spatial_size):
+    t = [
+        LoadImaged(keys=("image", "label"), reader="ITKReader"),
+        NormalizeLabelsInDatasetd(keys="label", label_names=labels),
+        AddChanneld(keys=("image", "label")),
+        Orientationd(keys=["image", "label"], axcodes="RAS"),
+        # This transform may not work well for MR images
+        ScaleIntensityRanged(keys="image", a_min=-175, a_max=250, b_min=0.0, b_max=1.0, clip=True),
+        RandFlipd(keys=("image", "label"), spatial_axis=[0], prob=0.10),
+        RandFlipd(keys=("image", "label"), spatial_axis=[1], prob=0.10),
+        RandFlipd(keys=("image", "label"), spatial_axis=[2], prob=0.10),
+        RandRotate90d(keys=("image", "label"), prob=0.10, max_k=3),
+        RandShiftIntensityd(keys="image", offsets=0.10, prob=0.50),
+        Resized(keys=("image", "label"), spatial_size=spatial_size, mode=("area", "nearest")),
+        # Transforms for click simulation
+        FindAllValidSlicesMissingLabelsd(keys="label", sids="sids"),
+        AddInitialSeedPointMissingLabelsd(keys="label", guidance="guidance", sids="sids"),
+        AddGuidanceSignalDeepEditd(keys="image", guidance="guidance"),
+        #
+        ToTensord(keys=("image", "label")),
+    ]
+
+    return Compose(t)
+
+
+def get_click_transforms():
+    t = [
+        Activationsd(keys="pred", softmax=True),
+        AsDiscreted(keys="pred", argmax=True),
+        ToNumpyd(keys=("image", "label", "pred")),
+        # Transforms for click simulation
+        FindDiscrepancyRegionsDeepEditd(keys="label", pred="pred", discrepancy="discrepancy"),
+        AddRandomGuidanceDeepEditd(
+            keys="NA",
+            guidance="guidance",
+            discrepancy="discrepancy",
+            probability="probability",
+        ),
+        AddGuidanceSignalDeepEditd(keys="image", guidance="guidance"),
+        #
+        ToTensord(keys=("image", "label")),
+    ]
+
+    return Compose(t)
+
+
+def get_post_transforms(labels):
+    t = [
+        Activationsd(keys="pred", softmax=True),
+        AsDiscreted(
+            keys=("pred", "label"),
+            argmax=(True, False),
+            to_onehot=(len(labels), len(labels)),
+        ),
+        # This transform is to check dice score per segment/label
+        SplitPredsLabeld(keys="pred"),
+    ]
+    return Compose(t)
+
+
+def get_loaders(args, pre_transforms, train=True):
```

Comment:
I don't see the usage of arg `train=False` in this code file, maybe review this arg directly?

Thanks.

Reply:
Agree, thanks!